idseq_dag/util/periodic_thread.py

```diff
@@ -0,0 +1,35 @@
+import threading
+import time
+
+class PeriodicThread(threading.Thread):
+    '''
+    Thread that keeps executing its target func periodically until it is told to stop.
+    Usage example:
+        t = PeriodicThread(target=poll_something, sleep_seconds=60, args=(some_argument_1, some_argument_2))
+        t.start()
+        ...
+        t.stop()
+        t.join()
+    '''
+
+    def __init__(self, target, sleep_seconds, args=(), kwargs=None):
+        super(PeriodicThread, self).__init__()
+        self.target = target
+        self.sleep_seconds = sleep_seconds
+        self.args = args
+        if kwargs is None:
+            self.kwargs = {}
+        else:
+            self.kwargs = kwargs
+        self._stop_event = threading.Event()
+
+    def stop(self):
+        self._stop_event.set()
+
+    def stopped(self):
+        return self._stop_event.is_set()
+
+    def run(self) -> None:
+        while not self.stopped():
+            self.target(*self.args, **self.kwargs)
```

Comment:
I would imagine the desired behavior here in the event the target throws an excaption is not to crash the entire thread, is it?    You probably want to catch the exception and record somehow that an exception has occurred at least once;   but that should not prevent future invocations of the target by crashing the thread, should it?

Reply:
thanks, changed